src/main/java/de/chojo/repbot/service/RoleAssigner.java

```diff
@@ -38,16 +42,27 @@ public void update(@Nullable Member member) {
 
         cleanMemberRoles(member);
 
+        assertInteract(roleById, member.getGuild());
+
         guild.addRoleToMember(member, roleById).queue();
     }
 
-    private void cleanMemberRoles(Member member) {
+    private void cleanMemberRoles(Member member) throws RoleAccessException {
         var guild = member.getGuild();
-        guildData.getReputationRoles(guild)
+        var collect = guildData.getReputationRoles(guild)
```

Comment:
"collect" doesn't describe this variable at all, "reputationRoles" would be a better choice